templates/react/src/libs/provider.ts

```diff
@@ -0,0 +1,110 @@
+import { ethers, providers, BigNumber } from 'ethers'
+import { Environment, CurrentConfig } from '../config'
+import { BaseProvider } from '@ethersproject/providers'
+
+// Single copies of provider and wallet
+const rpcProvider = new ethers.providers.JsonRpcProvider(CurrentConfig.rpc.mainnet)
+const localRpcProvider = new ethers.providers.JsonRpcProvider(CurrentConfig.rpc.local)
+
+const wallet = createWallet(localRpcProvider, rpcProvider)
+
+const windowProvider = createWindowProvider()
+let walletExtensionAddress: string | null = null
+
+// Interfaces
+
+export enum TransactionState {
+  Failed = 'Failed',
+  New = 'New',
+  Rejected = 'Rejected',
+  Sending = 'Sending',
+  Sent = 'Sent',
+}
+
+// Provider and Wallet Functions
+
+export function getMainnetProvider(): BaseProvider {
+  return rpcProvider
+}
+
+export function getProvider(): providers.Provider | null {
+  return CurrentConfig.env === Environment.WALLET_EXTENSION ? windowProvider : wallet.provider
+}
+
+export function getWalletAddress(): string | null {
+  return CurrentConfig.env === Environment.WALLET_EXTENSION ? walletExtensionAddress : wallet.address
+}
+
+export async function sendTransaction(transaction: ethers.providers.TransactionRequest): Promise<TransactionState> {
+  if (CurrentConfig.env === Environment.WALLET_EXTENSION) {
+    return sendTransactionViaExtension(transaction)
+  } else {
+    transaction.value = BigNumber.from(transaction.value)
+    return sendTransactionViaWallet(transaction)
+  }
+}
+
+export async function connectWallet() {
```

Comment:
This seems to refer to a web3 wallet whereas below wallet refers to an in code wallet.